src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -74,58 +58,40 @@ private void DynamicAllMonitor(TraceEvent obj)
-                    {
-                        HandleBeginInstrumentReporting(obj);
-                    }
-                    if (obj.EventName == "HistogramValuePublished")
-                    {
-                        HandleHistogram(obj);
-                    }
-                    else if (obj.EventName == "GaugeValuePublished")
-                    {
-                        HandleGauge(obj);
-                    }
-                    else if (obj.EventName == "CounterRateValuePublished")
-                    {
-                        HandleCounterRate(obj);
-                    }
-                    else if (obj.EventName == "UpDownCounterRateValuePublished")
-                    {
-                        HandleUpDownCounterValue(obj);
-                    }
-                    else if (obj.EventName == "TimeSeriesLimitReached")
-                    {
-                        HandleTimeSeriesLimitReached(obj);
-                    }
-                    else if (obj.EventName == "HistogramLimitReached")
+                    _renderer.SetErrorText(errorPayload.ErrorMessage);
+                    switch (errorPayload.ErrorType)
                     {
-                        HandleHistogramLimitReached(obj);
+                        case ErrorType.SessionStartupError:
+                            _shouldExit.TrySetResult(ReturnCode.SessionCreationError);
+                            break;
+                        case ErrorType.TracingError:
+                            _shouldExit.TrySetResult(ReturnCode.TracingError);
+                            break;
+                        case ErrorType.NonFatal:
+                            break;
+                        default:
+                            // Is this the behavior we want, or should we throw?
+                            _shouldExit.TrySetResult(ReturnCode.UnknownError);
+                            break;
                     }
-                    else if (obj.EventName == "Error")
-                    {
-                        HandleError(obj);
-                    }
-                    else if (obj.EventName == "ObservableInstrumentCallbackError")
-                    {
-                        HandleObservableInstrumentCallbackError(obj);
-                    }
-                    else if (obj.EventName == "MultipleSessionsNotSupportedError")
-                    {
-                        HandleMultipleSessionsNotSupportedError(obj);
-                    }
-                    else if (obj.EventName == "MultipleSessionsConfiguredIncorrectlyError")
+                }
+                else if (payload is CounterEndedPayload counterEnded)
+                {
+                    _renderer.CounterStopped(counterEnded);
+                }
+                else if (payload.IsMeter)
+                {
+                    MeterInstrumentEventObserved(payload.Provider, payload.Timestamp);
+                    if (payload is not InstrumentationStartedPayload)
```

Comment:
I'd suggest creating a boolean property or function that says if this event is a value published event:
```suggestion
                    if (payload.IsValuePublishedEvent)
                    {
                        ...
                    }
```

Its easier to read if the code shows what events it wants to process rather than what it is filtering out. Maintaining a filter also seems more likely to get inadvertently out-of-sync if we added a new event in the future.